Userland/Libraries/LibBitTorrent/Bencode/BDecoder.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2023, Pierre Delagrave <pierre.delagrave@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "BTypes.h"
+#include <AK/MemoryStream.h>
+
+namespace BitTorrent {
+
+// The 'bencoding' section of https://www.bittorrent.org/beps/bep_0003.html
+class BDecoder {
+
+public:
+    template<typename T>
+    static ErrorOr<T> parse(ReadonlyBytes& bytes)
```

Comment:
`[Readonly]Bytes` are small types, taking a reference to them is self-defeating, use a copy. (And a mutable reference is even less logical)